### PR TITLE
[Evpn Warmreboot] Added Dependancy check logic in VrfMgr

### PR DESCRIPTION
**Pull request by nkelapur**
**Added changes to handel dependency check in VrfMgr for warmreboot**

**Why I did it**

This was done to ensure for EVPN warm-reboot the order of data replay to kernel is maintained across various submodules and the kernel programming will be successful.

**How I verified it**

Verified with EVPN warmreboot

**Details if related**

More details in warmreboot section of EVPN VXLAN HLD
https://github.com/Azure/SONiC/pull/437/

### CODE REVIEW

**Comment by nkelapur** [comment]
Dependant on https://github.com/Azure/sonic-swss/pull/1276

**Comment by prsunny** [comment]
retest this please

**Comment by srj102** [comment]
retest this please

File: tests/test_warm_reboot.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def swss_check_RestoreCount(dvs, state_db, restore_count):
             if fv[0] == "restore_count":
                 assert int(fv[1]) == restore_count[key] + 1
             elif fv[0] == "state":
-                assert fv[1] == "reconciled"
+                assert fv[1] == "reconciled" or fv[1] == "disabled"
```
Comment (prsunny) [inline tests/test_warm_reboot.py:49]: Do we still need to check for "disabled"? If not, please remove

File: cfgmgr/intfmgr.cpp [hunk 2]
```diff
@@ -659,4 +699,13 @@ void IntfMgr::doTask(Consumer &consumer)
 
         it = consumer.m_toSync.erase(it);
     }
+    
+    if (replayNotDone  &&   WarmStart::isWarmStart() && m_pendingReplayIntfList.empty() )
```
Comment (prsunny) [inline cfgmgr/intfmgr.cpp:703]: Please remove extra space

File: cfgmgr/intfmgr.cpp [hunk 3]
```diff
@@ -626,6 +655,7 @@ bool IntfMgr::doIntfAddrTask(const vector<string>& keys,
 void IntfMgr::doTask(Consumer &consumer)
 {
     SWSS_LOG_ENTER();
+    static bool replayNotDone = true;
```
Comment (prsunny) [inline cfgmgr/intfmgr.cpp:658]: can you please rename to "replayDone" and set to false.

File: tests/test_warm_reboot.py (hunk 1 above)
Comment (nkelapur) [reply tests/test_warm_reboot.py:49]: Yes, we still need "disabled" to handle the cold restart scenario, specially for docker restart

File: cfgmgr/intfmgr.cpp (hunk 2 above)
Comment (nkelapur) [reply cfgmgr/intfmgr.cpp:703]: will do

File: cfgmgr/intfmgr.cpp (hunk 3 above)
Comment (nkelapur) [reply cfgmgr/intfmgr.cpp:658]: sure .. will change it

**Comment by nkelapur** [comment]
retest this please

**Comment by nkelapur** [comment]
retest this please

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
retest this please

**Comment by prsunny** [comment]
retest this please

**Comment by nkelapur** [comment]
retest this please

**Comment by srj102** [comment]
retest this please

**Comment by nkelapur** [comment]
retest this please

**Comment by nkelapur** [comment]
retest this please

**Comment by nkelapur** [comment]
retest this please

**Comment by nkelapur** [comment]
retest this please

**Comment by prsunny** [comment]
retest this please

**Comment by liushilongbuaa** [comment]
retest vs please

**Comment by prsunny** [comment]
retest this please

**Pull request merged**